Remotion/ObjectBinding/Web.Development.WebTesting.TestSite.Shared/Controls/BocListUserControl.ascx.cs

```diff
@@ -149,5 +154,25 @@ private BocListUserControlTestOutput TestOutput
     {
       get { return (BocListUserControlTestOutput)((Layout)Page.Master).GetTestOutputControl(); }
     }
+
+    protected void ValidationTestCaseRow (object sender, EventArgs e)
+    {
+      var jobs = ((Person)CurrentObject.BusinessObject!).Jobs;
+      AddValidationFailures(
+          JobList_Normal,
+          new ObjectValidationFailure(jobs[0], "Row validation failure message", "Localized row validation failure message"));
+    }
+
+    private void AddValidationFailures (BocList list, params ValidationFailure[] validationFailures)
+    {
+      list.FixedColumns.Insert(0, new BocValidationErrorIndicatorColumnDefinition());
+
+      var validationResult = BusinessObjectValidationResult.Create(new ValidationResult(validationFailures));
+
+      PrepareValidation();
+      FormGridManager.PrepareValidation();
```

Comment:
Please check with Web.Test. We should only need to use SmartPage.PrepareValidation()